core/src/main/java/org/apache/accumulo/core/spi/crypto/PerTableCryptoServiceFactory.java

```diff
@@ -46,12 +48,24 @@ public class PerTableCryptoServiceFactory implements CryptoServiceFactory {
   private static final TableId REC_FAKE_ID = TableId.of("RECOVERY_CryptoService_FAKE_ID");
 
   @Override
-  public CryptoService getService(CryptoEnvironment environment, Map<String,String> props) {
+  public CryptoService getService(CryptoEnvironment environment, Map<String,String> props)
+      throws CryptoException {
     if (environment.getScope() == CryptoEnvironment.Scope.WAL) {
-      return cryptoServiceMap.computeIfAbsent(WAL_FAKE_ID, (id) -> getWALServiceInitialized(props));
+      return cryptoServiceMap.computeIfAbsent(WAL_FAKE_ID, (id) -> {
+        try {
+          return getWALServiceInitialized(props);
+        } catch (CryptoException e) {
+          throw new UncheckedIOException("Error creating WAL crypto service", e);
```

Comment:
If getService() is already throwing CryptoException, why catch and rethrow as Unchecked?

Reply:
It's for the lambda, can't throw checked exceptions

Reply:
OK that makes sense. Check out my other comment about checked vs unchecked.